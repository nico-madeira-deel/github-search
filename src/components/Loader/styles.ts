import styled, { css } from 'styled-components'
import { LoaderProps } from '.'

export const Wrapper = styled.div<LoaderProps>`
  animation: spin 1s linear infinite;
  border-radius: 50%;
  height: 3rem;
  width: 3rem;

  @keyframes spin {
    to {
      transform: rotate(360deg);
    }
  }

  ${({ theme, isCenter }) => css`
    border: 5px solid ${theme.colors.blue};
    border-left-color: ${theme.colors.darkBlue};
    margin: ${theme.spacings.xsmall} ${isCenter ? 'auto' : '0'};
  `}
`
