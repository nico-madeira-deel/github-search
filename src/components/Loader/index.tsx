import * as S from './styles'

export type LoaderProps = {
  isCenter?: boolean
}

const Loader = ({ isCenter = false }: LoaderProps) => (
  <S.Wrapper data-testid="loader" isCenter={isCenter} />
)

export default Loader
