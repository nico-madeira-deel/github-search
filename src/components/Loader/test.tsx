import { renderWithTheme } from 'utils/tests/helpers'

import Loader from '.'

describe('<Loader />', () => {
  it('should render the loader properly', () => {
    const { container } = renderWithTheme(<Loader />)

    expect(container.firstChild).toMatchSnapshot()
  })
})
