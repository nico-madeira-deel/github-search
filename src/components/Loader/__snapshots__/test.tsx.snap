// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loader /> should render the loader properly 1`] = `
.c0 {
  -webkit-animation: spin 1s linear infinite;
  animation: spin 1s linear infinite;
  border-radius: 50%;
  height: 3rem;
  width: 3rem;
  border: 5px solid #89b0ae;
  border-left-color: #555b6e;
  margin: 1.6rem 0;
}

<div
  class="c0"
  data-testid="loader"
/>
`
